apps/rewards/contracts/RewardsCore.sol

```diff
@@ -119,11 +149,16 @@ contract RewardsCore is IsContract, AragonApp {
         uint balance;
         uint initialSupply = reward.referenceToken.totalSupplyAt(reward.blockStart);
         uint endingSupply = reward.referenceToken.totalSupplyAt(reward.blockStart + reward.duration);
+        supply = endingSupply - initialSupply;
+        if (supply == 0) {
+            return 0;
+        }
+
         uint initialBalance = reward.referenceToken.balanceOfAt(msg.sender, reward.blockStart);
         uint endingBalance = reward.referenceToken.balanceOfAt(msg.sender, reward.blockStart + reward.duration);
-        require(initialSupply < endingSupply, "The supply must have increased over the period");
-        require(initialBalance < endingBalance, "The user must have earned tokens over the period");
-        supply = endingSupply - initialSupply;
+        //require(initialSupply < endingSupply, "The supply must have increased over the period");
```

Comment:
Both of these requires should be valid, where were they causing issues?

Reply:
The reason these `requires` have no place in this function is that this is used as a view function by the frontend for the current user. We'll come back to these later